components/builder-minio/habitat/hooks/install

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+exec 2>&1
+set -euo pipefail
+
+# shellcheck disable=SC1091
+source "{{pkg.svc_config_install_path}}/minio-migration.sh"
+
+timestamp=$EPOCHSECONDS # provides a shared element in naming items below
+config_environment_for_migration "$timestamp"
+
+echo "Checking if MinIO migration is needed"
+if is_migration_from_removed_fs_backend_needed; then
+
+  echo "BEGIN MinIO migration"
+
+  if pgrep minio &>/dev/null; then
+
+    if ! minio_health_live_check; then
+      echo "There seems to be a MinIO process but it fails the health check"
+      exit 9
+    fi
+
+  else
+
+    minio_pkg_old="core/minio/2018-10-05T01-03-03Z/20181006010122"
+    # This pkg_id is the minio dependency used in the on-prem-stable version of
+    # https://bldr.habitat.sh/#/pkgs/habitat/builder-minio/7764/20181006010221
+    # which was on-prem-stable for 6+ years so this should be a safe choice
+
+    if ! hab pkg list $minio_pkg_old >/dev/null; then
+      echo "ERROR: $minio_pkg_old is not installed and we require it for the migration."
+      exit 11
+    fi
+
+    old_minio_stdout=$(mktemp -t "minio-old-stdout-$timestamp-XXXXXXXXXX")
+    hab pkg exec $minio_pkg_old minio -- server \
+      --config-dir "{{pkg.svc_config_path}}" \
+      /hab/svc/builder-minio/data |
+      tee "$old_minio_stdout" &
+
+    if ! minio_health_live_check; then
+      echo "MinIO is not running so MinIO migration cannot begin"
+      exit 10
+    fi
+
+  fi
+
+  minio_old_summary=$(summarize_old_minio_bucket_objects "$timestamp")
+
+  download_bucket_objects
+
+  minio_stop
+
+  MIGRATION_BACKUP_DIRECTORY=$(mktemp -d -t minio-data-backup-"$timestamp"-XXXXXXXXXX)
+  cp -r /hab/svc/builder-minio/data "$MIGRATION_BACKUP_DIRECTORY"
+  GLOBIGNORE=".:.."
+  for x in /hab/svc/builder-minio/data/*; do
```

Comment:
Inform user here that this might take some time.

Reply:
Added messaging.